logger/drain/factory.go

```diff
@@ -15,19 +14,14 @@ func NewDrain(drainURL string) (LogDrain, error) {
 		// nil means no drain-- which is valid
 		return nil, nil
 	}
-	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") {
-		drain, err := udp.NewDrain(drainURL)
-		if err != nil {
-			return nil, err
-		}
-		return drain, nil
-	}
-	if strings.HasPrefix(drainURL, "tcp://") {
-		drain, err := tcp.NewDrain(drainURL)
+	// Any of these three can use the same drain implementation
+	if strings.HasPrefix(drainURL, "udp://") || strings.HasPrefix(drainURL, "syslog://") || strings.HasPrefix(drainURL, "tcp://") {
```

Comment:
Got it. So it sounds like this factory function *is* the convenience abstraction for enabling the platform to simply request a new drain by passing in a URI string to one internal endpoint.